### PR TITLE
integration puzzles for model

### PR DESCRIPTION
**Pull request by h1alexbel**
Added the initial integration puzzles for puzzle prioritization module (`model`).
Documented an integration with UML sequence diagram: use of`LinearModel#predict(...)` inside original 0pdd.

ref: #532

### CODE REVIEW

**Comment by h1alexbel** [comment]
in this pull request I also added a `plantuml.yml` GitHub Action for diagram generation, I'm not sure do we really need it or not, but its there (I don't think we need to store just SVGs or PNGs).

Besides that we need to provide `EMAIL` and `USERNAME` of the person who will sign the commit with generated diagram.
I can put `yegor256@gmail.com` and `yegor256`.

**Comment by yegor256** [comment]
@h1alexbel the build fails. I guess, you need to merge `master` into your branch

**Comment by h1alexbel** [comment]
@yegor256 bracnhes merged

**Comment by h1alexbel** [comment]
@yegor256 reminder

**Comment by h1alexbel** [comment]
@yegor256 ping

**Comment by yegor256** [comment]
@pnatashap can you please review this pull request?

**Review by pnatashap — COMMENTED** [review]
Looks like it should be fixed

File: .github/workflows/plantuml.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      - name: Checkout Source
+        uses: actions/checkout@v4
+      - name: Generate SVG Diagrams
+        uses: holowinski/plantuml-github-action@main
+        with:
+          args: -v -tsvg diagrams/*.puml
+      - name: Commit changes
+        uses: EndBug/add-and-commit@v9
+        with:
+          author_name: ${{ secrets.USERNAME }}
```
[the remark this answers is not in the document]
Comment (h1alexbel) [reply .github/workflows/plantuml.yml]: @pnatashap fixed, take a look, please

**Comment by yegor256** [comment]
@rultor merger

**Comment by h1alexbel** [comment]
@yegor256 estimates fixed, try again please

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**